src/mongocxx/test/pool.cpp

```diff
@@ -141,13 +141,43 @@ TEST_CASE(
     instance::current();
 
 // GCC before 4.9.0 doesn't place max_align_t in the std namespace.
-#if defined(__clang__) || !defined(__GNUC__) || (__GNUC__ > 4) || \
-    (__GNUC__ == 4 && __GNUC_MINOR__ >= 9)
+//
+// TODO: If we ever need max_align_t somewhere else, then move this to
+// a private header somewhere and write a macro for it. Remember to
+// write push/pops for the macro in the private prelude.
+#if defined(__GLIBCXX__)
+
+#if defined(__clang__)
+
+#if __has_include(<shared_mutex>)
+    // Clang, with libstdc++, newer than 4.9.0
+    std::max_align_t dummy_address;
+#else
+    // Clang, with libstdc++, older than 4.9.0
+    max_align_t dummy_address;
+#endif
+
+#elif defined(__GNUC__)
+
+#if (__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 9)
+    // GCC, with libstdc++, newer than 4.9.0
     std::max_align_t dummy_address;
 #else
+    // GCC, with libstdc++, older than 4.9.0
     max_align_t dummy_address;
 #endif
 
+#else
+    // Using libstdc++, but compiler is neither clang nor GCC; assume
+    // we have it in the right place
+    std::max_align_t dummy_address;
+#endif
+
+#else
+    // Not using libstdc++, assume it is in the right place.
+    std::max_align_t dummy_address;
+#endif
+
```

Comment:
How do you feel about:

auto dummy_address = []{ using namespace std; return max_align_t{}; }();

That'll work regardless of where max_align_t is in std or not.  and it skips the preprocessor